packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -1725,8 +1725,33 @@ class Transform extends SingleChildRenderObjectWidget {
   /// The origin of the coordinate system (relative to the upper left corner of
   /// this render object) in which to apply the matrix.
   ///
+  /// IMPORTANT: This offset is applied AFTER any [alignment] transformation.
+  /// Since [alignment] defaults to [Alignment.center], the origin offset will be
+  /// relative to the center of the widget, not the top-left corner, unless
+  /// alignment is explicitly modified.
+  ///
+  /// Example of unexpected behavior with default alignment:
```

Comment:
I would say: "In this example, the origin appears to be offset from the center of the widget, since [alignment] defaults to [Alignment.center]."

Then remove the comment from the code.